index.js

```diff
@@ -50,6 +51,46 @@ client.on('message', message => {
         .send(MESSAGE_PREFIX + "Send '!verify' to access other channels")
         .catch(reason => console.log(reason))
     }
+    else if (text.startsWith(CONFIG.SELF_JOIN_COMMAND) || is_self_join_alias) {
+      let guild = client.guilds.get(CONFIG.GUILD_ID)
+      let msg = is_self_join_alias ? CONFIG.SELF_JOIN_ALIASES[text] : text
+      let msg_roles = msg.replace('!join', '').trim()
+
+      guild
+        .fetchMember(message.author)
+        .then(member => {
+          if (msg_roles === '') {
+            message.channel.send(`Please specify the role you would like to join`).catch(reason => console.log(reason))
+          } else {
+            let verified_role = role = guild.roles.find(role => role.name === CONFIG.VERIFIED_ROLE_NAME)
+            if (!CONFIG.SELF_JOIN_REQUIRE_VERIFIED || member.roles.has(verified_role.id)) {
+              for (let msg_role of msg_roles.split(',').map(role_name => role_name.trim())) {
+                const guild_role = guild.roles.find(role => role.name === msg_role)
+                if (guild_role) {
+                  if (CONFIG.SELF_JOIN_ROLE_NAMES.includes(msg_role)) {
+                    if (member.roles.has(guild_role.id)) {
+                      member
+                        .removeRole(guild_role)
+                        .then(message.channel.send(`Left role '${msg_role}'`).catch(reason => console.log(reason)))
```

Comment:
I don't like the idea of leaving a role with !join :/ 

Would it be too much work to make !leave for this purpose?

Reply:
and let people know they are already in the role instead

Reply:
Added a separate command for leaving, with appropriate logic to notify the user if they are already in the role and are trying to join, or if they are not in the role and trying to leave